mantidproject/mantid
# Get Correct Default Name for Summed Runs

**Description of work.**
Changed the way SANS' `AddRunsPresenter` calculates the default name for summed runs. This PR gets the correct number of leading 0s before the run number so the added file can be found by name just like any other run.

**Report to:** stephen.king@stfc.ac.uk

**To test:**
1. Interfaces -> SANS -> SANS v2
2. In the left hand tab bar, go to Sum runs
3. At the top of the page, enter comma separated run numbers e.g. 105476, 105476 and click Add
4. Notice the default name, it should have a number of leading 0s before the run number to make up to the correct instrument/facility number of digits
5. Click Sum
6. In mantid, click load and try to load using the default name WITHOUT the leading zeroes it added e.g. If I summed 105476, 105476, the default name would be LOQ00105476-add. In `Load` I would search for `LOQ105476-add`.
7. Mantid should be able to find the run
8. In sum runs, give one of the runs an instrument, e.g. 105476, LOQ105476. The default name should assume that you provided a full name so this will be used as is. In this example the default name will be LOQ105476-add.

Fixes #24491 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
